### PR TITLE
MTROPOLIS: Add support for 'Unit Re-Booted'

### PR DESCRIPTION
**Pull request by mduggan**
_Unit: Re-Booted_ is [music CD](https://en.wikipedia.org/wiki/Unit_(album)) with a data partition which includes an
mTropolis-based music video program.

This is not much of a game, but uses some variants of the headers and data
structures which are nice to support and should help with future projects.

Known issues:
* Palette for the menu is a bit broken, needs some work
* The bottom-most button in the menu still crashes trying to load a PlugInModifier version 2001

### CODE REVIEW

File: engines/mtropolis/data.cpp [hunk 1]
```diff
@@ -1195,6 +1230,9 @@ bool TypicalModifierHeader::load(DataReader& reader) {
 		|| !reader.readU16(lengthOfName))
 		return false;
 
+	if (loadV2Header && !reader.readU32(unknown5))
```
Comment (elasota) [inline engines/mtropolis/data.cpp:1233]: Is this applying to everything using TypicalModifierHeader?  (Or at least any case of it that you've found?)  If this is global for the whole project then I think it should be flagged in DataReader instead (like the project format already is).

Comment (mduggan) [reply engines/mtropolis/data.cpp:1233]: Yep, it happens with all the modifiers that exist in this project so I expect it's an all-or-nothing thing and the flag could be raised up to the reader.  I'll give that a go.

File: engines/mtropolis/boot.cpp [hunk 2]
```diff
@@ -1210,6 +1224,14 @@ const Game games[] = {
 		nullptr,
 		GameDataHandlerFactory<STTGSGameDataHandler>::create
 	},
+	// Unit: Rebooted
+	{
+		MTBOOT_UNIT_REBOOTED_WIN,
+		unitWinFiles,
+		unitWinDirectories,
+		nullptr,
+		GameDataHandlerFactory<STTGSGameDataHandler>::create
```
Comment (mduggan) [inline engines/mtropolis/boot.cpp:1233]: I should probably rename this. I think that the `AlbertGameDataHandler` is basically the same as `STTGSGameDataHandler`.   Maybe it they can be combined with a generic name for the handler that has a `StandardPlugIn`?

File: engines/mtropolis/data.cpp (hunk 1 above)
Comment (mduggan) [reply engines/mtropolis/data.cpp]: Done.

**Comment by elasota** [comment]
I think this is fine as-is for now.

**Comment by justinkb** [comment]
need to have a look at the changes in elements.cpp, getting segfaults at the obsidian intro video

https://github.com/scummvm/scummvm/pull/5378/files#diff-5d27dbabcdadb6804855707dba8bfea0f0f68c1592782e4d2a84030d42941b98R645 doesn't look right

**Comment by mduggan** [comment]
Oops, thanks for letting me know! I just pushed 704d1d02112 which fixes that.  I tested the fix on the Obsidian demo and it resolves the crash.  I don't have the full game, but it should be the same there.

**Comment by elasota** [comment]
I'll take a look at this some more when I get back to mTropolis development.  It looks like this is a mTropolis 2.0 project, so I want to try to identify the unknown fields first if possible.

**Pull request merged**